ts/src/index.ts

```diff
@@ -129,13 +130,28 @@ export async function start(config: Config = {}): Promise<void> {
         config);
     return;
   }
+  // start the v8 heap profiler if heap profiling is enabled. This must be done
+  // before any asynchronous calls are made so samples from the first tick can
+  // be captured.
+  if (!(config.disableHeap === false)) {
+    heapProfiler.startSamplingHeapProfiler(
+        config.heapIntervalBytes || defaultConfig.heapIntervalBytes,
+        config.heapMaxStackDepth || defaultConfig.heapIntervalBytes);
+  }
+
   let normalizedConfig: ProfilerConfig;
   try {
     normalizedConfig = await initConfig(config);
   } catch (e) {
+    heapProfiler.stopSamplingHeapProfiler();
     logError(`Could not start profiler: ${e}`, config);
     return;
   }
+
+  if (normalizedConfig.disableHeap) {
+    heapProfiler.stopSamplingHeapProfiler();
```

Comment:
Why is it stopped here? If this is sequence like "enable - disable - enable in the constructor" then it's pretty convoluted, any chance to simplify?

Reply:
I think I've simplified this.